test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/MonthDayTest.scala

```diff
@@ -0,0 +1,218 @@
+package org.scalajs.testsuite.javalib.time
+
+import java.time.chrono.IsoChronology
+import java.time.temporal.{UnsupportedTemporalTypeException, ValueRange, ChronoField}
+import java.time.{DateTimeException, LocalDate, Month, MonthDay}
+
+import org.junit.Assert._
+import org.junit.Test
+import org.scalajs.testsuite.utils.AssertThrows._
+
+/**
+  * Created by alonsodomin on 22/12/2015.
+  */
+class MonthDayTest extends TemporalAccessorTest {
+  import ChronoField._
+  import MonthDay._
+  import DateTimeTestUtil._
+
+  final val min = of(Month.JANUARY, 1)
+  final val max = of(Month.DECEMBER, 31)
+  final val leapMonth = of(Month.FEBRUARY, 29)
+
+  val samples = Month.values().map(month => (month, month.minLength(), month.maxLength())).flatMap {
+    case (month, minDay, maxDay) =>
+      if (minDay != maxDay) Seq(of(month, 1), of(month, minDay), of(month, maxDay))
+      else Seq(of(month, 1), of(month, minDay))
+  }.toSeq
+
+  val yearSamples = 2000 to 2020
+  val leapYearLifted = (year: Int) => IsoChronology.INSTANCE.isLeapYear(year)
+
+  override def isSupported(field: ChronoField): Boolean =
+    field == ChronoField.MONTH_OF_YEAR || field == ChronoField.DAY_OF_MONTH
+
+  @Test override def test_range(): Unit = {
```

Comment:
Just wondering how many _special_ cases would leak into that impl. `LocalDateTest`, for example, also overrides that method to add _a few_ special cases, not only for the `DAY_OF_MONTH` field but for other fields as well. Merging all the possible special cases in the parent class as more classes are added sounds to me like a maintenance nightmare.

If we want to keep it DRY then the base method should probably be refactored to consume some sort of function `TemporalField => ValueRange` that will return the expected range for that specific field in the concrete test implementation. That function could perfectly be a method in `TemporalTest` that implementors need to override to provide the special cases.

```scala
def expectedRangeFor(field: TemporalField): ValueRange = field.range

@Test def range(): Unit = {
  for {
    accessor <- samples
    field <- ChronoField.values
  } {
    if (accessor.isSupported(field)) {
      val expected = expectedRangeFor(field)
      assertEquals(expected, accessor.range(field))
    } else {
      expectThrows(classOf[UnsupportedTemporalTypeException], accessor.range(field))
    }
  }
}
```